crates/project/src/project.rs

```diff
@@ -2110,10 +2074,17 @@ impl Project {
                 cx.emit(Event::WorktreeRemoved(*id));
             }
             WorktreeStoreEvent::WorktreeOrderChanged => cx.emit(Event::WorktreeOrderChanged),
+            WorktreeStoreEvent::WorktreeUpdateSent(_) => {}
         }
     }
 
     fn on_worktree_added(&mut self, worktree: &Model<Worktree>, cx: &mut ModelContext<Self>) {
+        {
+            let mut remotely_created_models = self.remotely_created_models.lock();
+            if remotely_created_models.retain_count > 0 {
```

Comment:
What happens if after you've created a lock, the thread holding it panics and the data becomes invalid in some way? I don't know if it matters thought. 

https://doc.rust-lang.org/nomicon/poisoning.html

Maybe instead of locking, use a closure, like

```rust
self.remotely_created_models.update(|this| {
});
```
and roll back the changes if a panic is caught

Reply:
panics just tear down the app; so we don't need to worry.